mfr/extensions/image/templates/viewer.mako

```diff
@@ -1,4 +1,176 @@
+            ## and grabs the first image it finds to create the zoom image.
+            ##
+            ## `addSpan()` performs the "wrap" accordingly to cater for most cases
+            var addSpan = function () {
+
+                ## Only wrap the image once
+                if (wrapped) {
+                    return;
+                }
+
+                ## Obtain the original height and width for the image loaded
+                var baseImageHeight = parseInt(baseImage.css("height"), 10);
+                var baseImageWidth = parseInt(baseImage.css("width"), 10);
+
+                ## Detect the Google Chrome browser
+                var isChrome = !!window.chrome;
+
+                ## Detect if the image is downsized due to screen size
+                var isActualSize = heightLimit === baseImageHeight || widthLimit === baseImageWidth;
+
+                if (isActualSize || !isChrome) {
+                    ## Use the default wrapping suggested by http://www.jacklmoore.com/zoom/ if
+                    ## either of the two conditions below holds:
+                    ## 1.   Images are downsized but not in Google Chrome. Please see the flickering
+                    ##      issue mentioned below.
+                    ## 2.   Images are displayed in its actual size. No issue for all supported
+                    ##      browsers.
+                    baseImage.wrap("<div></div>").parent().css("display", "inline-block");
+                } else {
+                    ## Quirks: Chrome has a flickering bug when images are wrapped with `<div>` or
+                    ## `<span>`.  During zoom the scroll bar keeps appearing and disappearing which
+                    ## causes the image to keep resizing.  Wrapping with `<p>` instead to solve this
+                    ## annoying issue.
+                    baseImage.wrap("<p></p>")
+                }
+
+                wrapped = true;
+            };
+
+            var mouseZoom = function (event, delta) {
+
+                ## Disable page scroll when the mouse cursor are above the image
+                event.preventDefault();
+
+                if (delta > 0) {
+                    magScale += magStep;
+                    magScale = magScale > maxScale ? maxScale : magScale
+                } else {
+                    magScale -= magStep;
+                    magScale = magScale < minScale ? minScale : magScale
+                }
+
+                var image = $(".zoomImage");
+                magHeight = magHeight === null ? parseInt(image.css("height"), 10) : magHeight;
+                magWidth = magWidth === null ? parseInt(image.css("width"), 10) : magWidth;
+
+                image.css({
+                    width: magWidth * magScale,
+                    height: magHeight * magScale
+                });
+            };
+
+            ## Quirks: Need to use an in-memory copy of the image to prevent the zoom image from
```

Comment:
Added a detailed comments on why this in-memory copy of the base image is needed.